x-pack/plugin/profiling/src/test/java/org/elasticsearch/xpack/profiling/RestGetStackTracesActionTests.java

```diff
@@ -66,24 +66,37 @@ public void testPrepareParameterizedRequest() {
             assertThat(request, instanceOf(GetStackTracesRequest.class));
             GetStackTracesRequest getStackTracesRequest = (GetStackTracesRequest) request;
             assertThat(getStackTracesRequest.getSampleSize(), is(10_000));
+            assertThat(getStackTracesRequest.getRequestedDuration(), is(3_600.0d));
+            assertThat(getStackTracesRequest.getAwsCostFactor(), is(1.0d));
+            assertThat(getStackTracesRequest.getCustomCO2PerKWH(), is(0.005d));
+            assertThat(getStackTracesRequest.getCustomDatacenterPUE(), is(1.5d));
+            assertThat(getStackTracesRequest.getCustomPerCoreWattX86(), is(7.5d));
+            assertThat(getStackTracesRequest.getCustomPerCoreWattARM64(), is(2.0d));
+            assertThat(getStackTracesRequest.getCustomCostPerCoreHour(), is(0.083d));
             assertThat(getStackTracesRequest.getQuery(), notNullValue(QueryBuilder.class));
             executeCalled.set(true);
-            return new GetStackTracesResponse(
-                Collections.emptyMap(),
-                Collections.emptyMap(),
-                Collections.emptyMap(),
-                Collections.emptyMap(),
-                0,
-                0.0d,
-                0L
-            );
+
+            GetStackTracesResponseBuilder responseBuilder = new GetStackTracesResponseBuilder(getStackTracesRequest);
+            responseBuilder.setSamplingRate(0.04d);
+            responseBuilder.setTotalFrames(523);
+
+            GetStackTracesResponse response = responseBuilder.build();
+            assertEquals(new GetStackTracesResponse(null, null, null, null, 523, 0.04d, 0L), response);
```

Comment:
I'm a bit torn on this assert. Why do you prefer this to asserts of individual properties on the constructed response? As it is now, it's hard to understand what the individual properties mean without looking up the constructor signature (e.g. what does the `0L` mean?). How do you feel about asserting the individual properties?

Reply:
That is interesting, IntelliJ tells me the argument names, so I don't have that issue:
![Screenshot_20231206_093445](https://github.com/elastic/elasticsearch/assets/2087964/3d950fcc-ec17-46d8-bbbe-4601683370a8)

Reply:
I still think it would be preferable to check properties via individual asserts but I don't want to block that PR because of that. :) I'll approve.

Reply:
NP, testing each parameter with an own assert now.